src/cpu/intel.rs

```diff
@@ -25,9 +25,9 @@ mod abi_assumptions {
     const _ASSUMES_SSE2: () =
         assert!(cfg!(target_feature = "sse") && cfg!(target_feature = "sse2"));
 
-    #[cfg(target_arch = "x86_64")]
+    #[cfg(all(target_arch = "x86_64", target_pointer_width = "64"))]
     const _ASSUMED_POINTER_SIZE: usize = 8;
-    #[cfg(target_arch = "x86")]
+    #[cfg(any(target_arch = "x86", all(target_arch = "x86_64", target_pointer_width = "32")))]
```

Comment:
These assertions are here because we don't know if our code works with the x32 ABI.

At a minimum, we'd also need to duplicate all the x86-64 `coverage` tests in ci.yml for this new ABI, in addition to doing this.

This is because we're unsure whether *all* the assembly code correctly supports 32-bit `size_t` and 32-bit pointers.